### PR TITLE
cargo-binutils: init at 0.3.3

### PR DESCRIPTION
**Pull request by Stupremee**
###### Motivation for this change

[`cargo-binutils`](https://github.com/rust-embedded/cargo-binutils) contains various cargo subcommands to invoke the LLVM tools that are shipped with Rust.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Stupremee** [comment]
Result of `nixpkgs-review pr 108410` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>cargo-binutils</li>
  </ul>
</details>

File: pkgs/development/tools/rust/cargo-binutils/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+  src = let
+    repo = fetchFromGitHub {
+      owner = "rust-embedded";
+      repo = pname;
+      rev = "v${version}";
+      sha256 = "sha256-Dgn+f4aSsDSh+RC8yvt3ydkdtwib5jEVsnZkod5c7Vo=";
+    };
+  in runCommand "source" { } ''
+    cp -R ${repo} $out
```
Comment (doronbehar) [inline pkgs/development/tools/rust/cargo-binutils/default.nix:15]: Can you try to contact upstream regarding them not committing `Cargo.lock`?

**Review by doronbehar — CHANGES_REQUESTED** [review]
your PR should have eventually two commits:

```
maintainers: add <me>
<package-name>: init at <version>
```

File: pkgs/development/tools/rust/cargo-binutils/default.nix (hunk 1 above)
Comment (Stupremee) [reply pkgs/development/tools/rust/cargo-binutils/default.nix:15]: I've opened a [PR](https://github.com/rust-embedded/cargo-binutils/pull/99) :+1:

Comment (Stupremee) [reply pkgs/development/tools/rust/cargo-binutils/default.nix:15]: One of the maintainers of cargo-binutils [said](https://github.com/rust-embedded/cargo-binutils/pull/99#issuecomment-757503230) that they won't push the `Cargo.lock` file

File: pkgs/development/tools/rust/cargo-binutils/default.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
+{ stdenv, lib, fetchFromGitHub, rustPlatform, runCommand }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "cargo-binutils";
+  version = "0.3.3";
+
+  src = let
```
Comment (doronbehar) [inline pkgs/development/tools/rust/cargo-binutils/default.nix:7]:
```suggestion
  # Upstream doesn't commit `Cargo.lock`, see https://github.com/rust-embedded/cargo-binutils/pull/99
  src = let
```

File: pkgs/development/tools/rust/cargo-binutils/default.nix (hunk 1 above)
Comment (doronbehar) [reply pkgs/development/tools/rust/cargo-binutils/default.nix:15]: They ~can~ should use Nix :).

File: pkgs/development/tools/rust/cargo-binutils/default.nix [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    cp -R ${repo} $out
+    chmod -R +w $out
+    cp ${./Cargo.lock} $out/Cargo.lock
+  '';
+
+  cargoSha256 = "sha256-Zrl269PacPi81TrGTIDzmVndgGY5i5lYyspiOj43rpw=";
+
+  meta = with lib; {
+    description =
+      "Cargo subcommands to invoke the LLVM tools shipped with the Rust toolchain ";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/rust/cargo-binutils/default.nix:24]:
```suggestion
    description = "Cargo subcommands to invoke the LLVM tools shipped with the Rust toolchain ";
```

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108410` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>cargo-binutils</li>
</ul>
</details>

**Pull request merged**